src/k8s/pkg/client/kubernetes/status.go

```diff
@@ -20,6 +21,25 @@ func (c *Client) WaitKubernetesEndpointAvailable(ctx context.Context) error {
 	})
 }
 
+// IsClusterReady pings /readyz endpoint and only reports ready if status code == 200.
+func (c *Client) IsClusterReady(ctx context.Context) bool {
+	logger := log.FromContext(ctx)
+	result := c.Discovery().RESTClient().Get().AbsPath("/readyz").Do(ctx)
```

Comment:
this is hard to mock test with. we can stick with querying for the endpoints instead